From ac431b97de4ef0ec90bd0977305a1cce343c4bcc Mon Sep 17 00:00:00 2001
From: kk <kaiwalya.khasnis57@gmail.com>
Date: Wed, 14 Oct 2026 11:33:29 +0900
Subject: [PATCH 1/3] scripts/dtc: fix yylloc multiple definition on modern
 host GCC

GCC 10+ defaults to -fno-common, so the tentative definition of yylloc in
both the lexer and the parser breaks the host dtc build. Declare it extern
in both places instead.
---
 scripts/dtc/dtc-lexer.lex.c_shipped | 2 +-
 scripts/dtc/dtc-parser.y            | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/dtc/dtc-lexer.lex.c_shipped b/scripts/dtc/dtc-lexer.lex.c_shipped
index 2d30f41778b..d0eb405cb81 100644
--- a/scripts/dtc/dtc-lexer.lex.c_shipped
+++ b/scripts/dtc/dtc-lexer.lex.c_shipped
@@ -637,7 +637,7 @@ char *yytext;
 #include "srcpos.h"
 #include "dtc-parser.tab.h"
 
-YYLTYPE yylloc;
+extern YYLTYPE yylloc;
 
 /* CAUTION: this will stop working if we ever use yyless() or yyunput() */
 #define	YY_USER_ACTION \
diff --git a/scripts/dtc/dtc-parser.y b/scripts/dtc/dtc-parser.y
index f412460f94d..24ca3eb4256 100644
--- a/scripts/dtc/dtc-parser.y
+++ b/scripts/dtc/dtc-parser.y
@@ -24,7 +24,7 @@
 #include "dtc.h"
 #include "srcpos.h"
 
-YYLTYPE yylloc;
+extern YYLTYPE yylloc;
 
 extern int yylex(void);
 extern void print_error(char const *fmt, ...);
